test/functional/rpc_rawtransaction.py

```diff
@@ -69,24 +74,129 @@ def setup_network(self):
         super().setup_network()
         self.connect_nodes(0, 2)
 
+    def sync_peers(self):
+        self.sync_mempools(self.nodes[0:3])
+        self.sync_blocks()
+
+    def generate_and_sync(self, node, blocks):
+        self.sync_peers()
+        self.nodes[node].generate(blocks)
+        self.sync_peers()
+
     def run_test(self):
-        self.log.info('prepare some coins for multiple *rawtransaction commands')
-        self.nodes[2].generate(1)
-        self.sync_all()
-        self.nodes[0].generate(COINBASE_MATURITY + 1)
-        self.sync_all()
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),1.5)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),1.0)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),5.0)
-        self.sync_all()
-        self.nodes[0].generate(5)
-        self.sync_all()
-
-        self.log.info('Test getrawtransaction on genesis block coinbase returns an error')
+        self.log.info("Prepare some coins for multiple *rawtransaction commands")
+        self.generate_and_sync(node=2, blocks=1)
+        self.generate_and_sync(node=0, blocks=COINBASE_MATURITY + 1)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
+        self.generate_and_sync(node=0, blocks=5)
+
+        self.getrawtransaction_tests()
+        self.createrawtransaction_tests()
+        self.sendrawtransaction_tests()
+        self.sendrawtransaction_testmempoolaccept_tests()
+        self.decoderawtransaction_tests()
+        self.transaction_version_number_tests()
+        if not self.options.descriptors:
+            self.raw_multisig_transaction_legacy_tests()
+
+    def getrawtransaction_tests(self):
+        addr = self.nodes[1].getnewaddress()
+        txid = self.nodes[0].sendtoaddress(addr, 10)
+        self.generate_and_sync(node=0, blocks=1)
+        vout = find_vout_for_address(self.nodes[1], txid, addr)
+        rawTx = self.nodes[1].createrawtransaction([{'txid': txid, 'vout': vout}], {self.nodes[1].getnewaddress(): 9.999})
+        rawTxSigned = self.nodes[1].signrawtransactionwithwallet(rawTx)
+        txId = self.nodes[1].sendrawtransaction(rawTxSigned['hex'])
+        self.generate_and_sync(node=0, blocks=1)
+
+        # Make a tx by sending, then generate 2 blocks; block1 has the tx in it
+        tx = self.nodes[2].sendtoaddress(self.nodes[1].getnewaddress(), 1)
+        block1, block2 = self.nodes[2].generate(2)
+        self.sync_peers()
```

Comment:
Done

Reply:
Moving the in-mempool transaction tests commit before the simplification commit revealed that removing the `self.sync_peers()` would be flakey without the presence of the in-mempool transaction. This could be fragile, so no longer removing it in the simplification commit.